pkg/http/api_directmessaging.go

```diff
@@ -98,11 +98,8 @@ func (a *api) onDirectMessage(w http.ResponseWriter, r *http.Request) {
 	}
 	defer req.Close()
 
-	policyRunner := resiliency.NewRunnerWithOptions(
+	policyRunner := resiliency.NewRunner[*invokev1.InvokeMethodResponse](
 		r.Context(), policyDef,
-		resiliency.RunnerOpts[*invokev1.InvokeMethodResponse]{
-			Disposer: resiliency.DisposerCloser[*invokev1.InvokeMethodResponse],
```

Comment:
Sadly we do need this disposer, it's very important because otherwise we would not clean up HTTP responses (the disposer invokes `.Close()` on the response objects)

I'm not sure why this does the trick in your case, we can chat if you'd like! But the disposer has got to remain.

Reply:
We are invoking `.Close()`  lateron on the recieved response message: https://github.com/dapr/dapr/blob/725e7d6865e35726104b856208e2d60f1a583c75/pkg/http/api_directmessaging.go#L179
I understand still it won't close the other resp objects created during retries, but that is the only cleaner solution I can think of. Basically we need the response object in the end, to forward the received message and statuscode and disposer cleans that only. Do you have any other solution? We can chat for sure!

Reply:
Yes, we are invoking `Close()` later, but that's only invoked on the successful response.

The point of the disposer is that if a request fails with a timeout, the disposer is the only thing we have that performs a cleanup of the resources, which otherwise would leak.

Reply:
can we update this disposer to let it forward the http status code?

Reply:
No, the disposer isn’t aware of what the response is. Also the disposer is invoked on failed (like timed out) responses too

But the way to do this could be creating a custom Error object that contains the status code we need.

Reply:
Pushed the alternate implementation using a custom error object. Please review